clang/test/Driver/clang-offload-bundler-bc-archive-support2.cpp

```diff
@@ -0,0 +1,38 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Create BC file to be bundled
+// RUN: %clangxx -emit-llvm -c %s -o %t.bc
+// Create object file to be bundled
+// RUN: %clangxx            -c %s -o %t.o
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Bundle BC and object file to same target
+// RUN: clang-offload-bundler -type=bc -targets=host-x86_64-unknown-linux-gnu -input=%t.bc -output=%t_bundled.bc
+// RUN: clang-offload-bundler -type=o  -targets=host-x86_64-unknown-linux-gnu -input=%t.o  -output=%t_bundled.o
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Create unbundled BC and object file to use as reference
+// RUN: clang-offload-bundler -type=bc -targets=host-x86_64-unknown-linux-gnu -input=%t_bundled.bc -unbundle -output=%t_unbundled.bc
+// RUN: clang-offload-bundler -type=o  -targets=host-x86_64-unknown-linux-gnu -input=%t_bundled.o  -unbundle -output=%t_unbundled.o
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Make archive with bundled BC and o files
+// RUN: rm -f %t_bundled.a
+// RUN: llvm-ar cr %t_bundled.a %t_bundled.bc %t_bundled.o
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Test archive unbundling with multiple files with the same target
+// RUN: clang-offload-bundler -unbundle --targets=host-x86_64-unknown-linux-gnu -type=aoo -input=%t_bundled.a -output=%t_list.txt
+
+// RUN: wc %t_list.txt -l | FileCheck --check-prefixes=CHECK-LIST-LENGTH %s
+
+// CHECK-LIST-LENGTH: 2
+
+// RUN: cmp %t_unbundled.bc `cat %t_list.txt | grep .bc$`
+// RUN: cmp %t_unbundled.o  `cat %t_list.txt | grep .o$`
+
+#include <iostream>

```

Comment:
I think an empty file will also work here. If not, atleast an empty function.

Thanks

Reply:
Simplified test function.